crates/node/p2p/src/gossip/driver.rs

```diff
@@ -62,7 +71,7 @@ impl GossipDriver {
 
     /// Returns if the swarm has already successfully dialed the given [`Multiaddr`].
     pub fn has_dialed(&mut self, addr: &Multiaddr) -> bool {
-        self.dialed_peers.contains(addr)
+        self.dialed_peers.get(addr).cloned().unwrap_or(false)
```

Comment:
This doesn't make much sense at a glance - why is this better than `.contains` when `.get` doesn't take ownership?

Reply:
Because it's a hashmap and we want to get the value not if the key is contained.

But you call out a good point here that the `.cloned()` can be fixed to be `.copied()`.